torchmultimodal/models/mdetr.py

```diff
@@ -140,37 +132,38 @@ def forward(self, images: List[Tensor], text: List[Tensor]) -> MDETROutput:
         pos = self.pos_embed(image_mask).to(image_embeddings.dtype)
         query_embed = self.query_embed.weight
 
+        if self.extra_query_embeddings is not None:
+            n_extra_embeddings = self.extra_query_embeddings.num_embeddings
+            query_embed = torch.cat([query_embed, self.extra_query_embeddings.weight])
+
         text_memory_resized = self.text_projection(text_memory)
 
-        transformer_outputs = self.transformer(
+        transformer_output = self.transformer(
             self.image_projection(image_embeddings),
             image_mask,
             query_embed,
             pos,
             text_memory=text_memory_resized,
             text_attention_mask=text_attention_mask,
         )
-        final_hidden_state = transformer_outputs.decoder_hidden_states[-1]
+
+        if self.extra_query_embeddings is not None:
+            extra_embeddings = transformer_output.decoder_hidden_states[
+                0, :, -n_extra_embeddings:
+            ]
+            transformer_output = transformer_output._replace(
```

Comment:
this is confusing, could you maybe split this into two lines so it's more clear?